src/constructs/aws/Queue.ts

```diff
@@ -232,9 +232,19 @@ export class Queue extends AwsConstruct {
 
         const alarmEmail = configuration.alarm;
         if (alarmEmail !== undefined) {
+            // generate the display name, AWS restriction is 100 chars
+            let displayName = `[Alert][${id}] failed jobs in dlq.`;
+            if (displayName.length > 100) {
+                // if the length is too long then try to pretty print
+                displayName = `[Alert][${id.substring(
+                    0,
+                    id.length - (displayName.length - 100 - 3)
+                )}...] failed jobs in dlq.`;
+            }
```

Comment:
Thanks for the PR!

I think we can get rid of the `if` and always truncate: if the max length is not reached then we'll have the full name.

Reply:
The first option is what I meant 👍